core/jazz_admin/test/test.js

```diff
@@ -1,11 +1,171 @@
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+const expect = require('chai').expect;
+const sinon = require('sinon');
+const request = require('request');
+
 const index = require('../index');
+const awsContext = require('aws-lambda-mock-context');
+const configObj = require('../components/config.js');
+
+describe('jazz_admin', function () {
+
+  beforeEach(function () {
+    spy = sinon.spy();
+    event = {
+      "stage": "test",
+      "method": "GET",
+      "principalId" : "test@test.com",
+      "body":{}
+    };
+    callback = (err, responseObj) => {
+      if (err) {
+        return err;
+      } else {
+        return JSON.stringify(responseObj);
+      }
+    };
+    err = {
+      "errorType": "foo",
+      "message": "bar"
+    };
+    callbackObj = {
+      "callback": callback
+    };
+    config = configObj.getConfig(event, context);
+    context = awsContext();
+  });
+
+  describe('validation tests', () => {
+
+   it("should state that POST method is not supported", function () {
+      event.method = "POST";
+      index.handler(event, context, (error, data) => {
+        expect(error).to.include('{"errorType":"BadRequest","message":"The requested method is not supported"}');
+      });
+    });
+  
+    it("should state the user isn't authorized if no principalId is given", function () {
+      event.principalId = "";
+      index.handler(event, context, (error, data) => {
+        expect(error).to.include('{"errorType":"Unauthorized","message":"User is not authorized to access this service|Authorization Incomplete"}');
+      });
+    });
+
+    it("should state the user isn't authorized if principalId given is not same as admin id", function () {
+      event.principalId = "foo@foo.com";
+      index.handler(event, context, (error, data) => {
+        expect(error).to.include('{"errorType":"Unauthorized","message":"This user is not authorized to access this service.');
+      });
+    });
+  });
+
+  describe('getInstallerVarsJSON', () => {
+
+    it("should indicate error while making request to gitlab repo", () => {
+      config.SCM_TYPE = "gitlab";
+      var responseObj = {
+        statusCode: 401,
+        body: {
+          data: {},
+          message: "Unauthorized"
+        }
+      };
+      reqStub = sinon.stub(request, "Request").callsFake((obj) => {
+        return obj.callback(null, responseObj, responseObj.body)
+      });
+      index.getInstallerVarsJSON(config)
+        .catch(error => {
+          expect(error).to.include('Unauthorized');
+        });
+      sinon.assert.calledOnce(reqStub);

```

Comment:
you should put these statements(sinon.assert and restore inside the callback and not outside.

`.catch(error => {
          expect(error).to.include('Unauthorized');
          sinon.assert.calledOnce(reqStub);
          reqStub.restore();
  })`


fix it everywhere